autoload/lsp/completion.vim

```diff
@@ -114,22 +114,30 @@ enddef
 
 # add completion from current buf
 def CompletionFromBuffer(items: list<dict<any>>)
-    var words = {}
-    for line in getline(1, '$')
-        for word in line->split('\W\+')
-            if !words->has_key(word) && word->len() > 1
-                words[word] = 1
-                items->add({
-                    label: word,
-                    data: {
-                        entryNames: [word],
-                    },
-                    kind: 26,
-                    documentation: "",
-                })
-            endif
-        endfor
+  var words = {}
+  var start = reltime()
+  var linenr = 1
+  for line in getline(1, '$')
+    for word in line->split('\W\+')
+      if !words->has_key(word) && word->len() > 1
+	words[word] = 1
+	items->add({
+	  label: word,
+	  data: {
+	    entryNames: [word],
+	  },
+	  kind: 26,
+	  documentation: "",
+	})
+      endif
     endfor
+    # Check every 200 lines if timeout is exceeded
+    if opt.lspOptions.bufferCompletionTimeout > 0 && linenr % 200 == 0 && 
```

Comment:
For optimization reasons, can you store the value of opt.lspOptions.bufferCompletionTimeout in a local variable outside the for loop and then use that local variable here?  Otherwise, Vim will recompute the Dict index every time.  You can check the output of the `:disassemble` command for the difference in the number of instructions.

Reply:
I was thinking it will optimize dict key search, but it does not. There is LOADSCRIPT and USEDICT instructions inside for-loop. So you're correct.